### PR TITLE
fix: Fix/ruby install

### PR DESCRIPTION
**Pull request by Cal-L**
## **Description**

This PR fixes e2e build failures caused by a failed ruby install on Bitrise. This failure was a result of ruby being bumped from 3.1.4 -> 3.1.5 from Bitrise's changelog - https://stacks.bitrise.io/changelogs/osx-xcode-15.0.x/#stack-update-v2024-05-24-released-on-2024-05-27

## **Related issues**

Fixes:

## **Manual testing steps**

1. Recommend to use rbenv to manage ruby. If you need rbenv - `brew install rbenv`
2. Install new version of ruby - rbenv install 3.1.5
3. Set global version of ruby - rbenv global 3.1.5
4. Set local version of ruby - rbenv local 3.1.5
5. Re-run `yarn setup`
6. Should be able to compile apps as usual

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -200,6 +200,8 @@ workflows:
           inputs:
             - content: |-
                 #!/usr/bin/env bash
+                echo "Installing ruby version as specified in .ruby-version"
+                rbenv install $(cat .ruby-version)
```
Comment (sethkfman) [inline bitrise.yml:204]: I don't think this will be needed with the update.

Comment (Cal-L) [reply bitrise.yml]: Removed!

**Review by sethkfman — DISMISSED** [review]
LGTM

**Review by legobeat — DISMISSED** [review]

File: app/components/hooks/useAccounts/useAccounts.test.ts [hunk 2]
```diff
@@ -38,6 +38,7 @@ const MOCK_ACCOUNT_2: Account = {
 };
 
 const MOCK_STORE_STATE = {
+  settings: { showFiatOnTestnets: false },
```
Comment (NicolasMassart) [inline app/components/hooks/useAccounts/useAccounts.test.ts:41]: Not fond of having this fix here, but hell yes, that will be faster...
Add detail for this change in the PR description please.

File: ios/Podfile.lock [hunk 3]
```diff
@@ -1028,7 +1028,7 @@ EXTERNAL SOURCES:
 
 SPEC CHECKSUMS:
   Base64: cecfb41a004124895a7bcee567a89bae5a89d49b
-  boost: 57d2868c099736d80fcd648bf211b4431e51a558
+  boost: 7dcd2de282d72e344012f7d6564d024930a6a440
```
Comment (NicolasMassart) [inline ios/Podfile.lock:1031]: is this related to the ruby issue or something else?

File: app/components/hooks/useAccounts/useAccounts.test.ts (hunk 2 above)
Comment (NicolasMassart) [reply app/components/hooks/useAccounts/useAccounts.test.ts:41]: Given [this PR ](https://github.com/MetaMask/metamask-mobile/pull/9774) has been created, it may be cleaner to move your change to it?

File: ios/Podfile.lock (hunk 3 above)
Comment (Cal-L) [reply ios/Podfile.lock:1031]: It's not related and I've updated the PR description. Others will see this diff atm on main.

File: app/components/hooks/useAccounts/useAccounts.test.ts (hunk 2 above)
Comment (Cal-L) [reply app/components/hooks/useAccounts/useAccounts.test.ts]: Remove changes from here

File: ios/Podfile.lock (hunk 3 above)
Comment (Cal-L) [reply ios/Podfile.lock]: Removed changes from here

**Review by sethkfman — APPROVED** [review]
LGTM

**Review by NicolasMassart — APPROVED** [review]
Looks good to me.

**Comment by Cal-L** [comment]
Smoke test passes in earlier commit
